lib/src/test/kotlin/graphql/nadel/engine/NadelDeferSupportTest.kt

```diff
@@ -0,0 +1,208 @@
+        assertTrue(channel.isClosedForReceive)
+    }
+
+    @Test
+    fun `after last job the hasNext is false`() = runTest {
+        val channel = Channel<DelayedIncrementalPartialResult>(UNLIMITED)
+
+        val subject = NadelDeferSupport(channel)
+        val lockingJob = CompletableDeferred<Boolean>()
+
+        // When
+        val firstAsync = subject.defer {
+            DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult()
+                .incrementalItems(emptyList())
+                .hasNext(true)
+                .build()
+        }
+        val secondAsync = subject.defer {
+            DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult()
+                .incrementalItems(emptyList())
+                .hasNext(true)
+                .build()
+        }
+
+        subject.defer {
+            // Wait until test tells us to continue
+            lockingJob.join()
+
+            DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult()
+                .incrementalItems(
+                    listOf(
+                        DeferPayload.newDeferredItem()
+                            .data("Bye world")
+                            .path(listOf("echo"))
+                            .build()
+                    )
+                )
+                .hasNext(true)
+                .build()
+        }
+
+        subject.onInitialResultComplete()
+
+        // Then
+        firstAsync.join()
+        secondAsync.join()
+
+        lockingJob.complete(true)
+
+        val results = channel.consumeAsFlow().toList()
+        val lastResult = results.last()
+        assertTrue((lastResult.incremental?.single() as DeferPayload).getData<String>() == "Bye world")
+        assertFalse(lastResult.hasNext())
+    }
+
+    @Test
+    fun `hasNext is true if last job launches more jobs`() = runTest {
+        val channel = Channel<DelayedIncrementalPartialResult>(UNLIMITED)
+        val subject = NadelDeferSupport(channel)
+        val firstLock = CompletableDeferred<Boolean>()
+        val secondLock = CompletableDeferred<Boolean>()
+
+        // When
+        subject.defer {
+            // Wait until test tells us to continue
+            firstLock.join()
+
+            subject.defer {
+                secondLock.join()
+
+                DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult()
+                    .incrementalItems(emptyList())
+                    .hasNext(true)
+                    .build()
+            }
+
+            DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult()
+                .incrementalItems(emptyList())
+                .hasNext(false)
+                .build()
+        }
+
+        // Then
+        firstLock.complete(true)
+
+        val item = channel.receive()
+        assertTrue(item.hasNext())
+    }
+
+    @Test
+    fun `hasNext is true if there is another job still running`() = runTest {
+        // Channel that stores the oldest item
```

Comment:
I can't understand what this comment means 🤔 There's only one channel where all the items are published to, right?

Reply:
Oh yeah that comment seems outdated, I'll remove it. I think before I had `Channel(1, DROP_OLDEST)` or something, but I changed it.